File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    # Use a dictionary to translate one nucleotide to another.
    transcribe = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U',
    }
    
    # Loop over the dna strand building the result.
    # Throw an exception if you find a value not in the transcribe dictionary.
    result = ''
    for ch in dna_strand:
        if ch in transcribe:
            result += transcribe[ch]
        else:
            raise Exception('Invalid nucleotide.')
    return result

File: csharp/rna-transcription/RnaTranscription.cs
using System.Linq;
public static class RnaTranscription
{
    public static string ToRna(string strand)
    {
        Dictionary<char, char> translate = new Dictionary<char, char> {
            ['G'] = 'C',
            ['C'] = 'G',
            ['T'] = 'A',
            ['A'] = 'U'
        };

        return new string((
            from char ch in strand 
            select translate.GetValueOrDefault(ch, ch)).ToArray<char>());
    }
}